### ISSUE TITLE
Homebrew formula whoozle-android-file-transfer-nightly is broken

### ISSUE THREAD

**Issue by cpcallen** [whoozle/android-file-transfer-linux#264]
The `-nightly` brew formula is broken because the `.dmg` it tries to download does not exist:

```
$ brew install homebrew/cask-versions/whoozle-android-file-transfer-nightly
==> Downloading https://github.com/whoozle/android-file-transfer-linux/releases/download/continuous/AndroidFileTransferForLinux.dmg

curl: (22) The requested URL returned error: 404 
Error: Download failed on Cask 'whoozle-android-file-transfer-nightly' with message: Download failed: https://github.com/whoozle/android-file-transfer-linux/releases/download/continuous/AndroidFileTransferForLinux.dmg
```

Maybe not strictly an AFTL problem, but since this formula is mentioned in the `README.md`, I thought I should bring it to your attention.

**Comment by LeeBinder** [comment]
still..

> Last login: Wed Jan  8 21:27:47 on ttys000

`/Users/lee:$brew install --cask whoozle-android-file-transfer-nightly`

> Error: Download failed on Cask 'whoozle-android-file-transfer@nightly' with message: Download failed: https://github.com/whoozle/android-file-transfer-linux/releases/download/continuous/AndroidFileTransferForLinux.dmg
> /Users/lee:$

**Comment by whoozle** [comment]
not still, it broke very recently :rofl:

**Comment by whoozle** [comment]
can you try again please? I think I fixed nightly builder so it picks up new brew directory structure

**Comment by LeeBinder** [comment]
Thank you @whoozle , the command now again works, but it downloads the Apple Silicon aarch64 version even though I'm on an Intel Mac :/ Has the Intel fork also been compiled by the nightly builder, but something is missing to point to it when running the brew command on an Intel Mac?

**Comment by whoozle** [comment]
oh dear, they must've been some change in GH CI builders then

**Comment by LeeBinder** [comment]
All the versions before, both release and nightly, have been Intel 64 bit. From what I know from other macOS projects here on GitHub, there must be *separate* build scripts for Intel and aarch64/ Apple Silicon.

Can you manually produce an Intel artifact via Actions?

**Comment by whoozle** [comment]
it seems "macos-latest" resolves to arm64 builds now 
https://github.blog/changelog/2024-01-30-github-actions-macos-14-sonoma-is-now-available/

**Comment by LeeBinder** [comment]
I see .. and only `macos` still to Intel?

**Comment by LeeBinder** [comment]
yes, in general, even though many people still use Intel Macs which are 100% fine for most everyday tasks. I'll keep mine for as long as possible, even if as as 2ndary Mac.

If you need a "nudge" on how to tackle this: [Romain Vimont](https://github.com/rom1v) from Genymobile just recently managed this with his scrcpy app https://github.com/Genymobile/scrcpy If you want to, I can send you a link to the topic there.

**Comment by whoozle** [comment]
the main problem is that I rely on homebrew to install deps and they are not / won't be supporting universal binaries. This means to have universal build I have to build all deps myself and drop brew stuff completely. 

This is something to consider given how hacky building macosx is currently done. Building all deps with cmake also allows proper bundling right out of the box at least.

**Comment by whoozle** [comment]
it looks like downgrading builder just worked, let me try qt6 upgrade as well...

**Comment by LeeBinder** [comment]
That's great news. And yes exactly - Romain Vimont also uses cmake for his scrcpy, and it compiles perfectly for both macOS architectures.

**Comment by whoozle** [comment]
all green, should be good to install now!

**Comment by LeeBinder** [comment]
Yes, and the app is working - well done and thank you 👍 

Now you also know how to make a native aarch64 build. Are you going to implement that somehow for the growing Apple Silicon crowd, maybe by having Artifacts created in the Actions section here?

**Comment by LeeBinder** [comment]
BTW, for brew you could use separate casks, one for Intel and one for aarch:

brew install --cask whoozle-android-file-transfer-nightly
brew install --cask whoozle-android-file-transfer@nightly
